pkg/network/usm/ebpf_ssl.go

```diff
@@ -545,8 +558,34 @@ func (o *sslProgram) GetStats() *protocols.ProtocolStats {
 	return nil
 }
 
-func addHooks(m *manager.Manager, probes []manager.ProbesSelector) func(utils.FilePath) error {
+func isBuildKit(procRoot string, pid uint32) bool {
+	filePath := filepath.Join(procRoot, strconv.Itoa(int(pid)), "comm")
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		// Waiting a bit, as we might get the event of process creation before the directory was created.
+		for i := 0; i < 3; i++ {
+			time.Sleep(10 * time.Millisecond)
+			// reading again.
+			content, err = os.ReadFile(filePath)
+			if err == nil {
+				break
+			}
```

Comment:
Not blockers to the PR but:
* We could avoid one allocation if we used `os.Open/File.Read` instead (it also looks like like `FileRead` issues an extra syscall (fstat) that we don't really need.
* Should we decrease the 10ms backoff? My gut feeling is that this could be too long and we could miss part of the SSL socket creation bits, but it's hard to say if this will actually impact or not;
* Did you run this against the load test environment?

Reply:
Addressed your first 2 comments
re: load test, I'll run it tomorrow after the pipeline will be finished

Reply:
> Addressed your first 2 comments

Sort of 😅 
`io.ReadAll` has the same problem of `os.ReadFile` in that it doesn't take a buffer argument. I was suggesting to use a plain [File.Read](https://pkg.go.dev/os#File.Read) instead, but that means you would have to add a buffer pool as well.

Given that this is not really in the hot-loop of process events because we already filter down the things that trigger call `addHooks` I think it's fine to leave it as it is, but just something to keep in mind in case this pops up in the profiling data.

Reply:
@p-lambert let me know if now it is better

Reply:
There is actually one last thing missing 😅 
Right now you may leave "garbage" data in the buffer that is re-used, so either you clean the 16 bytes or you trim the buffer like `buf[:n]` using the return value from `Read`

Reply:
done!